File: backend/src/main/java/com/back/back9/domain/user/repository/UserRepository.java
package com.back.back9.domain.user.repository;

import com.back.back9.domain.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.List;

public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByApiKey(String apiKey);
    Optional<User> findByUserLoginId(String userLoginId);
    Optional<User> findByUsername(String username);
    List<User> findByUsernameContaining(String keyword);
}

File: backend/src/main/kotlin/com/back/back9/domain/user/repository/UserRepository.kt
package com.back.back9.domain.user.repository

import com.back.back9.domain.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, Long> {
    fun findByApiKey(apiKey: String): User?
    fun findByUserLoginId(userLoginId: String): User?
    fun findByUsername(username: String): User?
    fun findByUsernameContaining(keyword: String): List<User>
}